mlflow/utils/request_utils.py

```diff
@@ -187,7 +187,7 @@ def _get_http_response_with_retries(
     session = _get_request_session(
         max_retries, backoff_factor, backoff_jitter, retry_codes, raise_on_status
     )
-    return session.request(method, url, **kwargs)
+    return session.request(method, url, allow_redirects=False, **kwargs)
```

Comment:
nit: Wondering if we want to show some warning/error when getting 302? As it seems impossible to really know the impact of this change, we/users may want to know if anything is broken, instead of just getting raw response with 302.

Reply:
Yeah I can add a catch if the response status code is 30x and print a warning!

Reply:
@B-Step62 it looks like printing a warning will only print it on the server side, but it's not guaranteed that the user will have access to server logs (e.g. they might just be interacting with the server API responses). i think it's still better than nothing, at least whoever is running the server will have some indication of the redirect to investigate, but i think editing response metadata might be a bit too big of a change. what do you think?

Reply:
That's true..... I think warning works as a minimum option and agree that modifying response is not preferable, but how about even raising an exception? I think silent failure is the worse case.

Reply:
Does `raise_for_status` throw if the status code is 30x?

Reply:
It doesn't.